game/scripts/vscripts/items/azazel_tower_watch.lua

```diff
@@ -45,99 +49,65 @@ item_azazel_tower_watch_4 = item_azazel_tower_watch_1
 
 modifier_watch_tower = class(ModifierBaseClass)
 
-local SINK_HEIGHT = 200
 local THINK_INTERVAL = 0.1
 
 function modifier_watch_tower:OnCreated()
   local ab = self:GetAbility()
   self.level = ab:GetLevel()
-  if IsServer() then
-    local target = self:GetParent()
-    local maxhealth = ab:GetSpecialValueFor("health")
-    local location = target:GetOrigin()
-    local time = ab:GetSpecialValueFor("construction_time")
-    target:Attribute_SetIntValue("construction_time", time)
-    target:Attribute_SetIntValue("bonus_damage", ab:GetSpecialValueFor("bonus_damage"))
-    target:Attribute_SetIntValue("bonus_vision_range", ab:GetSpecialValueFor("bonus_vision_range"))
-    target:SetOrigin(GetGroundPosition(location, target) - Vector(0, 0, SINK_HEIGHT))
-    Timers:CreateTimer(0.5, function()
-      target:RemoveModifierByName('modifier_invulnerable')
-      ResolveNPCPositions(location, target:GetHullRadius())
-      target:SetMaxHealth(maxhealth)
-      target:SetHealth(maxhealth * 0.01)
-      target:SetMaterialGroup('radiant_level'..self.level)
-      self:StartIntervalThink(THINK_INTERVAL)
-      self:SetStackCount(math.floor(time / THINK_INTERVAL)) -- `construction_time` should be divisible by `THINK_INTERVAL`!
-    end)
-  end
+  self.bonus_vision_range = ab:GetSpecialValueFor("bonus_vision_range")
+  self:StartIntervalThink(THINK_INTERVAL)
 end
 
 function modifier_watch_tower:OnIntervalThink()
   if IsServer() then
     local target = self:GetParent()
-    local count = self:GetStackCount()
-    if count > 0 then
-      local time = target:Attribute_GetIntValue("construction_time", 10)
-      target:SetOrigin(target:GetOrigin() + Vector(0, 0, SINK_HEIGHT / (time / THINK_INTERVAL)))
-      self:SetStackCount(count - 1)
-    else
-      -- No other way to have unobstructed vision, sorry.
-      AddFOWViewer(target:GetTeam(), target:GetOrigin(), target:GetCurrentVisionRange(), THINK_INTERVAL, false)
-    end
+    -- No other way to have unobstructed vision, sorry.
+    AddFOWViewer(target:GetTeam(), target:GetOrigin(), target:GetCurrentVisionRange(), THINK_INTERVAL, false)
```

Comment:
The building isn't supposed to provide any vision while constructing. Your version does that.

Reply:
@ZzZombo You are almost a year late. Azazel buildings are not in the game anymore.